instrumentation/hystrix-1.4/javaagent/src/test/java/io/opentelemetry/javaagent/instrumentation/hystrix/HystrixTest.java

```diff
@@ -0,0 +1,181 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.javaagent.instrumentation.hystrix;
+
+import static io.opentelemetry.api.common.AttributeKey.booleanKey;
+import static io.opentelemetry.api.common.AttributeKey.stringKey;
+import static io.opentelemetry.sdk.testing.assertj.OpenTelemetryAssertions.equalTo;
+import static io.opentelemetry.sdk.testing.assertj.OpenTelemetryAssertions.satisfies;
+import static io.opentelemetry.semconv.trace.attributes.SemanticAttributes.EXCEPTION_STACKTRACE;
+import static io.opentelemetry.semconv.trace.attributes.SemanticAttributes.EXCEPTION_TYPE;
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.junit.jupiter.api.Named.named;
+
+import com.netflix.hystrix.HystrixCommand;
+import com.netflix.hystrix.HystrixCommandGroupKey;
+import com.netflix.hystrix.HystrixCommandProperties;
+import io.opentelemetry.instrumentation.testing.junit.AgentInstrumentationExtension;
+import io.opentelemetry.instrumentation.testing.junit.InstrumentationExtension;
+import io.opentelemetry.sdk.trace.data.StatusData;
+import java.util.Objects;
+import java.util.concurrent.BlockingQueue;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.LinkedBlockingQueue;
+import java.util.function.Function;
+import java.util.stream.Stream;
+import org.junit.jupiter.api.extension.RegisterExtension;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.Arguments;
+import org.junit.jupiter.params.provider.MethodSource;
+
+class HystrixTest {
+
+  @RegisterExtension
+  protected static final InstrumentationExtension testing = AgentInstrumentationExtension.create();
+
+  @ParameterizedTest
+  @MethodSource("provideCommandActionArguments")
+  void testCommands(Function<HystrixCommand<String>, String> operation) {
+
+    HystrixCommand<String> command =
+        new HystrixCommand<String>(setter()) {
+          @Override
+          protected String run() throws Exception {
+            return tracedMethod();
+          }
+
+          private String tracedMethod() {
+            testing.runWithSpan("tracedMethod", () -> {});
+            return "Hello!";
+          }
+        };
+
+    String result = testing.runWithSpan("parent", () -> operation.apply(command));
+    assertThat(Objects.equals(result, "Hello!")).isTrue();
+
+    testing.waitAndAssertTraces(
+        trace ->
+            trace.hasSpansSatisfyingExactly(
+                span -> span.hasName("parent").hasNoParent(),
+                span ->
+                    span.hasName("ExampleGroup.HystrixTest$1.execute")
+                        .hasParent(trace.getSpan(0))
+                        .hasAttributesSatisfyingExactly(
+                            equalTo(stringKey("hystrix.command"), "HystrixTest$1"),
```

Comment:
IDK whether this is worth the effort but instead of anonymous classes could use method local classes, something like
```java
    class TestCommand extends HystrixCommand<String> {

      protected TestCommand(Setter setter) {
        super(setter);
      }

      @Override
      protected String run() throws Exception {
        return tracedMethod();
      }

      private String tracedMethod() {
        testing.runWithSpan("tracedMethod", () -> {});
        return "Hello!";
      }
    }

    HystrixCommand<String> command = new TestCommand(setter());
```
then here this assertion would be `equalTo(stringKey("hystrix.command"), "TestCommand")` Then for each test in the same file would need to use a different name for the command to avoid `$x` in the class name.

Reply:
great idea, thank you!